joblib/numpy_pickle_utils.py

```diff
@@ -299,7 +299,7 @@ def __init__(self, filename, mode="rb", compresslevel=9):
         else:
             raise ValueError("Invalid mode: %r" % (mode,))
 
-        if isinstance(filename, (str, bytes)):
+        if isinstance(filename, (str, bytes, (sys.version_info < (3,) and unicode) or str)):
```

Comment:
Hmm, maybe better to do 

```py
from ._compat import _basestring

if isinstance(filename, _basestring):
```

to be more consistent with other places in the code.